indra/belief/evidence_subtype_tagger.py

```diff
@@ -0,0 +1,76 @@
+from __future__ import absolute_import, print_function, unicode_literals
+from builtins import dict, str
+from future.utils import python_2_unicode_compatible
+import pickle
+import sys
+import logging
+import re
+
+logger = logging.getLogger('evidence_subtype_tagger')
+
+reach_rule_filename = 'data/reach_rule_substrings.p'
+try:
+    reach_rule_regexp = pickle.load( open( reach_rule_filename, "rb" ) )
+except:
+    logger.error('Cannot open ', reach_rule_filename, ', will not be able' + 
+            ' to use priors for individual reach rules')
+
+def determine_reach_subtype(evidence):
+    """Returns the subtype of the reach rule. Looks at a list of regular
+    expressiosn corresponding to reach rule types, and returns the longest regexp that matches, or None if none of them match.
+    
+    Parameters
+    ----------
+    evidence: indra.statements.Evidence
+        A reach evidence object to subtype
+    """
+
+    assert(evidence.source_api == 'reach')
+    event_name = evidence.annotations['found_by']
+
+    best_match_length = None
+    best_match = None
+    for ss in reach_rule_regexp:
+        if re.search(ss, event_name):
+            if best_match is None or len(ss) > best_match_length:
+                best_match = ss
+                best_match_length = len(ss)
+
+    return best_match
```

Comment:
I would actually move this function into indra.sources.reach.processor, and change the API to take the event_name as an argument. The Belief Engine could then just import this function and call it with evidence.annotations['found_by'] to get the subtype.